### PR TITLE
Add array pluck

### PR DESCRIPTION
**Pull request by ameech**
Handy feature to grab specific keys from arrays.

### CODE REVIEW

File: src/array.php [hunk 1]
```diff
@@ -59,3 +59,24 @@ function to_array($value)
...
+ *
+ * @param array $source
+ * @param string $key
+ *
+ * @return array
+ */
+function pluck(array $source, $key)
+{
+    $array = array_map(function($array) use($key) {
+        if (isset($array[$key])) {
```
Comment (shadowhand) [inline src/array.php:74]: This will never return `null` values but I think that's okay.

File: src/array.php [hunk 2]
```diff
@@ -59,3 +59,24 @@ function to_array($value)
...
+ * NULL values are removed from results
+ *
+ * @param array $source
+ * @param string $key
+ *
+ * @return array
+ */
+function pluck(array $source, $key)
+{
+    $array = array_map(function($array) use($key) {
```
Comment (shadowhand) [inline src/array.php]: This violates PSR-2 style, it should be `function (...) use (...) {`

File: tests/ArrayTest.php [hunk 3]
```diff
@@ -93,4 +93,15 @@ public function testToArray($value)
         // And with traversables
         $this->assertSame($value, Equip\to_array($traversable));
     }
+
+    public function testPluck()
+    {
+        $source = [
+            ['id' => 1, 'company' => 'Acme'],
+            ['id' => 2, 'company' => 'Umbrella']
```
Comment (shadowhand) [inline tests/ArrayTest.php]: Trailing comma?

File: src/array.php (hunk 2 above)
Comment (ameech) [inline src/array.php]: I always confuse that one, thanks!

File: tests/ArrayTest.php (hunk 3 above)
Comment (ameech) [inline tests/ArrayTest.php]: Whoops.

File: src/array.php (hunk 1 above)
Comment (ameech) [inline src/array.php:74]: I thought so too, but: https://3v4l.org/D01pO

File: src/array.php [hunk 4]
```diff
@@ -59,3 +59,24 @@ function to_array($value)
...
+ */
+function pluck(array $source, $key)
+{
+    $array = array_map(function ($array) use ($key) {
+        if (isset($array[$key])) {
+            return $array[$key];
+        }
+    }, $source);
+
+    return array_filter($array);
```
Comment (shadowhand) [inline src/array.php]: I don't think this part is desired or wanted, really. It will nuke `0` values, for instance.

Comment (shadowhand) [inline src/array.php]: And if I do want that behavior, I can call `array_filter(pluck(...))` ezpz.

Comment (ameech) [inline src/array.php]: You're right, updated.

**Pull request merged**